src/renderer/RawRenderer.ts

```diff
@@ -2,10 +2,202 @@
+
+    // Resize the newly added panels
+    newPanels.forEach(panel => panel.resize());
+
+    // Update camera & control
+    camera.updateRange();
+    camera.updateAnchors();
+    camera.resetNeedPanelHistory();
+    control.updateInput();
+
+    // Move to the first panel added if no panels existed
+    // FIXME: fix for animating case
+    if (newPanels.length > 0 && !control.animating) {
+      void control.moveToPanel(control.activePanel || newPanels[0], 0).catch(() => void 0);
+    }
+
+    return newPanels;
+  }
+
+  public remove(index: number, deleteCount: number = 1): ElementPanel[] {
+    const panels = this._panels;
+    const flicking = getFlickingAttached(this._flicking, "Renderer");
+
+    const { camera, control } = flicking;
+    const removingIdx = getMinusCompensatedIndex(index, panels.length);
+
+    const panelsPulled = panels.slice(removingIdx + deleteCount);
+    const panelsRemoved = panels.splice(removingIdx, deleteCount);
+
+    // Update panel indexes
+    panelsPulled.forEach(panel => panel.decreaseIndex(panelsRemoved.length));
+
+    // Reset the order of the elements
+    this.resetPanelElementOrder();
+
+    // Remove panel elements
+    this._removePanelElements(panelsRemoved);
+
+    // Update camera & control
+    camera.updateRange();
+    camera.updateAnchors();
+    camera.resetNeedPanelHistory();
+    control.updateInput();
+
+    if (includes(panelsRemoved, control.activePanel)) {
+      control.resetActivePanel();
+    }
+
+    // FIXME: fix for animating case
+    if (panelsRemoved.length > 0 && !control.animating) {
+      const activePanel = control.activePanel;
+      const targetPanel = includes(panelsRemoved, activePanel) ? panelsPulled[0] || panels[panels.length - 1] : activePanel;
+
+      if (targetPanel) {
+        void control.moveToPanel(targetPanel, 0).catch(() => void 0);
+      } else {
+        // All panels removed
+        camera.lookAt(0);
+      }
+    }
+
+    return panelsRemoved;
+  }
+
+  public movePanelElementsToStart(panels: ElementPanel[]) {
+    const flicking = getFlickingAttached(this._flicking, "Renderer");
+    const camera = flicking.camera;
+    const cameraElement = camera.element;
+    const camRangeDiff = camera.rangeDiff;
+
+    const panelEls = panels.map(panel => panel.element);
+
+    const refElement = includes(panelEls, cameraElement.firstElementChild)
+      ? null
+      : cameraElement.firstElementChild;
+    this._relocatePanelElements(panels, refElement);
+
+    panels.forEach(panel => {
+      panel.decreaseOffset(camRangeDiff);
+    });
+
+    return this;
+  }
+
+  public movePanelElementsToEnd(panels: ElementPanel[]) {
+    const flicking = getFlickingAttached(this._flicking, "Renderer");
+    const camera = flicking.camera;
+    const camRangeDiff = camera.rangeDiff;
+
+    this._relocatePanelElements(panels, null);
+
+    panels.forEach(panel => {
+      panel.increaseOffset(camRangeDiff);
+    });
+
+    return this;
+  }
```

Comment:
nit) It seems to be similar logic. How about doing something like this?

```ts
private movePanelElements(panels: ElementPanel[], refElement, method){
    const flicking = getFlickingAttached(this._flicking, "Renderer");
    const camera = flicking.camera;
    const camRangeDiff = camera.rangeDiff;

    this._relocatePanelElements(panels, refElement);
    panels.forEach(panel => {
      panel[method](camRangeDiff);
    });

    return this;

}

  public movePanelElementsToStart(panels: ElementPanel[]) {
    const flicking = getFlickingAttached(this._flicking, "Renderer");
    const camera = flicking.camera;
    const cameraElement = camera.element;

    const panelEls = panels.map(panel => panel.element);
    const refElement = includes(panelEls, cameraElement.firstElementChild)
      ? null
      : cameraElement.firstElementChild;

    return this.movePanelElements(panels, refElement,"decreaseOffset");

  }

  public movePanelElementsToEnd(panels: ElementPanel[]) {
    return this.movePanelElements(panels,null,"increaseOffset");
  }
```

Reply:
Well, codes are changed a bit from that point...